atlasdb-config/src/main/java/com/palantir/atlasdb/factory/TransactionManagers.java

```diff
@@ -608,6 +615,34 @@ private TransactionManager serializableInternal(@Output List<AutoCloseable> clos
         return transactionManager;
     }
 
+    @SuppressWarnings("unused") // todo(rhuffman): wire up in a follow up PR
+    private KeyValueSnapshotReaderManager createKeyValueSnapshotReaderManager(
+            TransactionKeyValueServiceManager transactionKeyValueServiceManager,
+            TransactionService transactionService,
+            SweepStrategyManager sweepStrategyManager,
+            MetricsManager metricsManager) {
+        Optional<KeyValueSnapshotReaderManagerFactory> serviceDiscoveredFactory = config().transactionKeyValueService()
+                .map(AtlasDbServiceDiscovery::createKeyValueSnapshotReaderManagerFactoryOfCorrectType);
+        DeleteExecutor deleteExecutor = DefaultDeleteExecutor.createDefault(
+                transactionKeyValueServiceManager.getKeyValueService().orElseThrow());
+        OrphanedSentinelDeleter orphanedSentinelDeleter =
+                new DefaultOrphanedSentinelDeleter(sweepStrategyManager::get, deleteExecutor);
```

Comment:
It's a relatively arbitrary choice to create the orphanedSentinelDeleter here, especially because the deleteExecutor is still passed down. We could pass the sweepStrategyManager instead and construct the orphanedSentinelDeleter at a lower level.

Defer to you for what makes the most sense today, but this is something I'd anticipate we'll refactor during the sweep work. With that in mind the KVSRMF api is the most annoying to break, so if we're happy with the OrphanedSentinelDeleter interface being part of that api, we should leave as-is.

Reply:
Yeah this evaluation of the tradeoffs makes sense. I think let's not include the orphaned sentinel deleter at this layer (it feels too much like an implementation detail) and just have the Default one create it?

Reply:
Actually if we don't pass in the OrphanedSentinelDeleter here, we still need to pass in the SweepStrategyManager, and this now leaks a bunch of the Sweep strategy implementation details into the api module.

Reply:
Ah, yeah then that's bad. Agree, SweepStrategyManager _really_ shouldn't be passed in from here